### DISCUSSION TITLE
How can I run Calculated Values in Javascript with ketcher version 2.3.3?

### DISCUSSION THREAD

**Discussion by horiu** [epam/ketcher#1116] [Q&A] [answered]
Hi,

thanks for developing Ketcher!

Using the Calculated Values feature of the ketcher editor, I can calculate Chemical Formula and Exact Mass. This is a very nice feature. If I want to run this in Javascript and get the values, how can I do that?
I wish I could do it as a function, like ketcher.getSmilesAsync( ) and ketcher.generateImageAsync( ).

Thanks,

**Comment by AndreiMazol** [comment]
Hi @horiu ,
Sorry for delay  I missed your message.
You  can use `ketcher.server` object for this goal in  release 2.3.3:
`ketcher.server.calculate({your data here})`.
Unfortunately this functionality has been hidden in scope of release 2.4.0, so ketcher API will be reworked in the next release in scope of https://github.com/epam/ketcher/issues/1172.

Best regards,
Andrei

**Comment by horiu** [comment, marked as answer]
Thank you for your answer.
I'm looking forward to Release 2.5.0!

**Reply by AndreiMazol** [reply]
Hi @horiu ,

As a follow-up to my previous answer I'd like to  share one more option how to  use server functions mentioned above without Ketcher itself. You can use[ indigo-ketcher ](https://www.npmjs.com/package/indigo-ketcher) package directly for this purpose which contains all server-side functions used by ketcher.

Best regards,
Andrei
